### PR TITLE
REFACTOR-#4093: Refactor base to be smaller 

### LINKED ISSUE #4093: pandas/base.py refactor to be smaller
[pandas/base.py](https://github.com/modin-project/modin/blob/master/modin/pandas/base.py) is at 3809 files, and the four classes `BasePandasDataset`, `Resampler`, `Window`, and `Rolling` don't all need to be in base.py. We should move them to their own .py files to make this file more manageable.

### PR DESCRIPTION
**Pull request by jeffreykennethli**
## What do these changes do?

Split out `Resampler`, `Window`, and `Rolling` classes from `modin.core.pandas.base`.

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #4093? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [ ] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -2073,6 +2073,8 @@ def resample(
         origin: Union[str, TimestampConvertibleTypes] = "start_day",
         offset: Optional[TimedeltaConvertibleTypes] = None,
     ):
+        from .resampler import Resampler
```
Comment (anmyachev) [inline modin/pandas/base.py:2076]: Why are `Resampler`, `Window`, `Rolling` objects imported inside methods and not at the module level?

Comment (jeffreykennethli) [reply modin/pandas/base.py:2076]: Rebased. Thank you!

File: modin/core/storage_formats/base/doc_utils.py [hunk 2]
```diff
@@ -321,7 +321,7 @@ def doc_reduce_agg(method, refer_to, params=None, extra_params=None):
 
         {build_rules}
     """,
-    refer_to_module_name="DataFrame.Resampler",
+    refer_to_module_name="Resampler",
```
Comment (dchigarev) [inline modin/core/storage_formats/base/doc_utils.py:324]:
```suggestion
    refer_to_module_name="resample.Resampler",
```
For the actual path to the method:
```python
>>> # Before applying the suggestion
>>> print(BaseQueryCompiler.resample_nunique.__doc__)
...
Notes
-----
Please refer to ``modin.pandas.Resampler.nunique`` for more information
about parameters and output format.
>>>
>>> # After applying the suggestion
>>> print(BaseQueryCompiler.resample_nunique.__doc__)
Notes
-----
Please refer to ``modin.pandas.resample.Resampler.nunique`` for more information
about parameters and output format.

```

File: modin/core/storage_formats/base/doc_utils.py [hunk 3]
```diff
@@ -334,7 +334,7 @@ def doc_resample_reduce(result, refer_to, params=None, compatibility_params=True
     result : str
         The result of the method.
     refer_to : str
-        Method name in ``modin.pandas.base.Resampler`` module to refer to for
+        Method name in ``modin.pandas.Resampler`` module to refer to for
```
Comment (dchigarev) [inline modin/core/storage_formats/base/doc_utils.py:337]:
```suggestion
        Method name in ``modin.pandas.resample.Resampler`` module to refer to for
```

File: modin/core/storage_formats/base/doc_utils.py [hunk 4]
```diff
@@ -550,7 +550,7 @@ def doc_window_method(
     result : str
         The result of the method.
     refer_to : str
-        Method name in ``modin.pandas.base.Window`` module to refer to
+        Method name in ``modin.pandas.Window`` module to refer to
```
Comment (dchigarev) [inline modin/core/storage_formats/base/doc_utils.py:553]:
```suggestion
        Method name in ``modin.pandas.window.Window`` module to refer to
```

**Comment by dchigarev** [comment]
@jeffreykennethli please add a release note, otherwise lgtm!

File: modin/core/storage_formats/base/doc_utils.py (hunk 2 above)
Comment (jeffreykennethli) [reply modin/core/storage_formats/base/doc_utils.py]: Done, thank you.

File: modin/core/storage_formats/base/doc_utils.py (hunk 4 above)
Comment (jeffreykennethli) [reply modin/core/storage_formats/base/doc_utils.py]: Done, thank you.

**Comment by jeffreykennethli** [comment]
Release notes added, thanks!

**Review by mvashishtha — APPROVED** [review]
LGTM! Thanks for the refactor.

**Review by dchigarev — DISMISSED** [review]
LGTM, will merge on green CI

**Review by devin-petersohn — APPROVED** [review]
@jeffreykennethli Please fix merge conflict, LGTM otherwise!

**Pull request merged**